### PR TITLE
python3Packages.anyio: disable all socket-related tests on darwin

### PR DESCRIPTION
**Pull request by mweinelt**
###### Motivation for this change

@SuperSandro2000 #99521 

They don't work in their sandbox.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/anyio/default.nix [hunk 1]
```diff
@@ -1,4 +1,5 @@
-{ lib
+{ stdenv
+, lib
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/anyio/default.nix:2]: I am feeling a bit nit picky here but stdenv and lib are not really both required.

Comment (mweinelt) [reply pkgs/development/python-modules/anyio/default.nix:2]: Though I am using both directly, so I don't think it's bad style per se.

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 102193` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages failed to build:</summary>
  <ul>
    <li>python37Packages.anyio</li>
    <li>python37Packages.ircrobots</li>
    <li>python38Packages.anyio</li>
    <li>python38Packages.ircrobots</li>
  </ul>
</details>

**Comment by mweinelt** [comment]
Pushed a version disabling the correct file. Please retest.

**Comment by mweinelt** [comment]
> Pushed a version disabling the correct file. Please retest.

@SuperSandro2000

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 102193` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python37Packages.anyio</li>
    <li>python37Packages.ircrobots</li>
    <li>python38Packages.anyio</li>
    <li>python38Packages.ircrobots</li>
  </ul>
</details>

**Pull request merged**